### PR TITLE
Minimal fixes to make pygame bootstrap work with python2legacy

### PR DESCRIPTION
**Pull request by opacam**
The pygame bootstrap is not working with the latest changes in master branch:

  -  because the old python2 recipe has been moved to python2legacy and the pygame bootstrap was made for python2legacy recipe (the android paths for the new python2 are different...so...it's unlikely that it works without touching anything)
  - The java build method used for pygame bootstrap is ant (which has been deprecated in modern android ndks and the default java version is not anymore in linux distribution packages (1.5)...now we need version 1.7)
- The distributed jpeg library in sdl does not build fine with modern ndks (assembler issues)

So...saying that, this pr fixes all the mentioned issues and makes that the pygame bootstrap uses the refactored distribution methods (already in python recipe, no need to touch that). 

This has been tested with ndk-r17c, ndk-api=19, android-api=27 and arch `armeabi-v7a`.

This is a quick fix because **still will be issues to solve...pyjnius** is not working at runtime, despite it compiles/installs fine (pygame bootstrap uses the old `renpy` entries instead of the `kivy` ones...).

**Build Note**: because we still use ant build for pygame bootstrap, the path of the final apk is not the same as the ones used by other bootstraps (the others uses gradle)...so...buildozer may have troubles with that (I dont know that for sure...it's only my guess)

**Note about pyjnius**: I suspect that we can make pyjnius work by moving from ant to gradle (and move the relevant code from renpy folder to the kivy one)...but this is quite a work...I'm not sure if we should do that...

I know that has been some talks **about removing pygame...maybe it's the time...but even if we do that...this code does not affect the core of the project...** so I think that could be safely merged after the proper reviews/discussions because all the changes introduced in here only affects files needed by pygame/sdl...so...easy to remove later.

### CODE REVIEW

**Comment by opacam** [comment]
**Update:** this pr may fix the issue described in #1582

File: pythonforandroid/recipes/pygame_bootstrap_components/__init__.py [hunk 1]
```diff
@@ -25,5 +25,11 @@ def prebuild_arch(self, arch):
         info('Unpacking was successful, deleting original container dir')
         shprint(sh.rm, '-rf', self.get_build_dir(arch))
 
+        info('Applying jpeg assembler patch')
+        ndk_15_plus_patch = join(self.get_recipe_dir(), 'jpeg-ndk15-plus.patch')
+        shprint(sh.patch, '-t', '-d',
+                join(self.get_build_container_dir(arch), 'jpeg'), '-p1',
+                '-i', ndk_15_plus_patch, _tail=10)
```
Comment (AndreMiras) [inline pythonforandroid/recipes/pygame_bootstrap_components/__init__.py]: Just wondering why don't we apply the patch using the `patches` attribute?

**Comment by AndreMiras** [comment]
Looking good to me. I'm just curious about why would you use `pygame` over `sdl2`?
The Travis conditional build is of course failing because it doesn't support other bootstraps yet, see https://github.com/kivy/python-for-android/issues/1588

Error was: `Didn't find any valid dependency graphs. This means that some of your requirements pull in conflicting dependencies.`

File: pythonforandroid/recipes/pygame_bootstrap_components/__init__.py (hunk 1 above)
Comment (opacam) [reply pythonforandroid/recipes/pygame_bootstrap_components/__init__.py]: As far as I remember...it's for the jpeg path, If we use the default patching method the file will not be found...but I'm thinking now that maybe we can add the needed levels to the patch, so we could refactor this into the default patching mechanism...I suppose that this should work...I will try to make some tests about that ;)

Comment (AndreMiras) [reply pythonforandroid/recipes/pygame_bootstrap_components/__init__.py]: Yes that would be awesome, that would simplify the code a little bit :+1:

**Pull request merged**